### ISSUE TITLE
Department Directory - acronyms are lowercase

### ISSUE THREAD

**Issue by coreywoodfield** [byu-oit-appdev/android-byu-suite-v2#148]
Not sure how we'll go about telling if something is an acronym, but all the acronyms are indiscriminately title cased. They probably shouldn't be. See the pic in #147 for an example (the title in the toolbar)

**Comment by romrell4** [comment]
Here is the code used to titleCase in iOS:

```objective c
-(NSString *)stringByApplyingBYUTitleCapitalization {
    static NSArray *wordsWithCapitalizationExceptions;
    
    if(!wordsWithCapitalizationExceptions) {
        wordsWithCapitalizationExceptions = @[
                                              @"and",
                                              @"BYU",
                                              @"BYUSA",
                                              @"CAEDM",
                                              @"CB",
                                              @"CIO",
                                              @"CITES",
                                              @"CES",
                                              @"CNA",
                                              @"CTB",
                                              @"DNA",
                                              @"FLAS",
                                              @"FM",
                                              @"for",
                                              @"ICLA",
                                              @"ID",
                                              @"KBYU",
                                              @"LDS",
                                              @"McDonald",
                                              @"McKay",
                                              @"MTC",
                                              @"MPH",
                                              @"N7BYU",
                                              @"NMELRC",
                                              @"of",
                                              @"ORCA",
                                              @"RB",
                                              @"ROTC",
                                              @"SAS",
                                              @"SFH",
                                              @"TEC",
                                              @"the",
                                              @"TV",
                                              @"WSC",
                                              @"YNews",
                                              @"YSA",
                                              ];
    }
    
    NSString *toReturn = [self stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
    toReturn = toReturn.capitalizedString;
    
    for (NSString *word in wordsWithCapitalizationExceptions) {
        NSString *regexPattern = [NSString stringWithFormat:@"\\b%@\\b", word];
        NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:regexPattern options:NSRegularExpressionCaseInsensitive error:nil];
        
        toReturn = [regex stringByReplacingMatchesInString:toReturn options:0 range:NSMakeRange(0, toReturn.length) withTemplate:word];
    }
    
    // capitalize first letter in the string, even if was changed to lowercase because of an exception word
    toReturn = [toReturn stringByReplacingCharactersInRange:NSMakeRange(0,1) withString:[[toReturn substringToIndex:1] uppercaseString]];

    return toReturn;
}
```

**Comment by romrell4** [comment]
Also, make sure that words after slashes (Air Conditioning/heating And Refrigeration) are capitalized. Should be Air Conditioning/Heating and Refrigeration.

**Comment by romrell4** [comment]
And hyphens (Anthropology-archaeology should be Anthropology-Archaeology)

**Comment by coreywoodfield** [comment]
I added "OIT" and also "in" and "www.". I also wrote an alternate implementation for titleCase, if you want to look and see which one you'd prefer that'd be great. I wrote the alternate implementation hoping I'd be able to figure out how to integrate all those exceptions into the loop we already had but I was unable to figure out a reasonable solution for how to do that unless I used a negative lookahead in the regex for any of those options... which might actually be a good idea. I'll probably look into that tomorrow
